From 97b1938efab057e7d3724bc0cc2dcab51444e246 Mon Sep 17 00:00:00 2001
From: Chen Meng J <meng.j.chen@intel.com>
Date: Fri, 17 May 2019 10:35:39 +0800
Subject: [PATCH 1484/1548] media: intel-ipu4: ox03a10: fix llp and fll ctrl
 for ox03a10

fix llp and fll ctrl for ox03a10.

Change-Id: I53df2b9ccfdf6fa50c0bdc030ae64b4a0cf15201
Tracked-On: PKT-2588
Tracked-On: #JIIAP-799
Signed-off-by: Chen Meng J <meng.j.chen@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 .../media/i2c/crlmodule/crl_ox03a10_common.h  | 38 +++++++++++++++++--
 1 file changed, 34 insertions(+), 4 deletions(-)

diff --git a/drivers/media/i2c/crlmodule/crl_ox03a10_common.h b/drivers/media/i2c/crlmodule/crl_ox03a10_common.h
index 6f3fedfefec2..f18382582c79 100644
--- a/drivers/media/i2c/crlmodule/crl_ox03a10_common.h
+++ b/drivers/media/i2c/crlmodule/crl_ox03a10_common.h
@@ -1274,6 +1274,36 @@ static struct crl_flip_data ox03a10_flip_configurations[] = {
 	},
 };
 
+static struct crl_dynamic_register_access ox03a10_llp_regs[] = {
+	{
+		.address = 0x380c,
+		.len = CRL_REG_LEN_08BIT,
+		.ops_items = ARRAY_SIZE(bits_15_8),
+		.ops = bits_15_8,
+	},
+	{
+		.address = 0x380d,
+		.len = CRL_REG_LEN_08BIT,
+		.ops_items = ARRAY_SIZE(bits_7_0),
+		.ops = bits_7_0,
+	}
+};
+
+static struct crl_dynamic_register_access ox03a10_fll_regs[] = {
+	{
+		.address = 0x380e,
+		.len = CRL_REG_LEN_08BIT,
+		.ops_items = ARRAY_SIZE(bits_15_8),
+		.ops = bits_15_8,
+	},
+	{
+		.address = 0x380f,
+		.len = CRL_REG_LEN_08BIT,
+		.ops_items = ARRAY_SIZE(bits_7_0),
+		.ops = bits_7_0,
+	}
+};
+
 struct crl_v4l2_ctrl ox03a10_v4l2_ctrls[] = {
 	{
 		.sd_type = CRL_SUBDEV_TYPE_PIXEL_ARRAY,
@@ -1509,8 +1539,8 @@ struct crl_v4l2_ctrl ox03a10_v4l2_ctrls[] = {
 		.flags = V4L2_CTRL_FLAG_UPDATE,
 		.impact = CRL_IMPACTS_NO_IMPACT,
 		.ctrl = 0,
-		.regs_items = ARRAY_SIZE(ar0231at_llp_regs),
-		.regs = ar0231at_llp_regs,
+		.regs_items = ARRAY_SIZE(ox03a10_llp_regs),
+		.regs = ox03a10_llp_regs,
 		.dep_items = 0,
 		.dep_ctrls = 0,
 		.v4l2_type = V4L2_CTRL_TYPE_INTEGER,
@@ -1529,8 +1559,8 @@ struct crl_v4l2_ctrl ox03a10_v4l2_ctrls[] = {
 		.flags = V4L2_CTRL_FLAG_UPDATE,
 		.impact = CRL_IMPACTS_NO_IMPACT,
 		.ctrl = 0,
-		.regs_items = ARRAY_SIZE(ar0231at_fll_regs),
-		.regs = ar0231at_fll_regs,
+		.regs_items = ARRAY_SIZE(ox03a10_fll_regs),
+		.regs = ox03a10_fll_regs,
 		.dep_items = 0,
 		.dep_ctrls = 0,
 		.v4l2_type = V4L2_CTRL_TYPE_INTEGER,
-- 
2.20.1

